## 引言

经过[前文](https://golfxiao.blog.csdn.net/article/details/142333893)对数据的校正与增强后，我们的预期生成结果中不再仅仅是分类标签，还多了欺诈者和分类原因。这样之前[模型评测](https://golfxiao.blog.csdn.net/article/details/141355995)和[批量评测](https://golfxiao.blog.csdn.net/article/details/141609627)两篇文章所封装的evaluate.py脚本就不再满足，需要对脚本进行改造，以支持新输出内容的评测。

新的预期结果中共包含三个信息，由于三个信息的特点不同，需要为每个字段制定不同的评测方式：
- is_fraud: 属于二分类，继续采用精确率和召回率作为评测指标。
- fraud_speaker: 属于多分类，多分类没有混淆矩阵，可以使用sklearn包提供的accuracy_score计算准确率作为评测指标。
- reason: 属于分文生成，由于没有准确值，可以通过rouge_score计算文本串的相似度作为评测指标。

## Rouge

ROUGE（Recall-Oriented Understudy for Gisting Evaluation）指标是用于评估文本摘要质量的一种常用指标。它通过比较生成的摘要与参考摘要之间的重叠词语或短语来衡量它们之间的相似度。Rouge包含多个子指标：
- ROUGE-1： 评估生成文本（hypothesis）与参考文本（reference）之间的单词级别的重叠； 
- ROUGE-2： 评估生成文本和参考文本相邻两个词（或字）的重叠情况，它关注的是两个文本中二元组（bigrams）的匹配，以此类推可以推广到ROUGE-N； 
- ROUGE-L：与前两者不同的是，此指标关注的是生成文本与参考文本之间的最长公共子序列（Longest Common Subsequence, LCS），通过最长公共子序列来衡量两个文本之间的相似性，最长公共子序列是指在不改变顺序的情况下，两个序列中都存在的元素的最长序列。

下面来实测一下。

In [1]:
from rouge import Rouge

def rouge_score(predictions, realitys):
    rouge = Rouge()
    return rouge.get_scores(predictions, realitys, avg=True)

先用英文测试。

In [10]:
predictions = "you are a programer."
realitys = "you are the programer."
scores = rouge_score(predictions, realitys)
scores

{'rouge-1': {'r': 0.75, 'p': 0.75, 'f': 0.749999995},
 'rouge-2': {'r': 0.3333333333333333,
  'p': 0.3333333333333333,
  'f': 0.3333333283333334},
 'rouge-l': {'r': 0.75, 'p': 0.75, 'f': 0.749999995}}

- 用rouge-1衡量时，predictions和realitys中有[`you`, `are`, `programer`]三个单词相同，生成文本和参考文本都是共4个单词，所以精确率和召回率都是75%；
- 用rouge-2衡量时，predictions和realitys中只有`you are`一个二元组相同，生成文本和参考文本都有4个单词（即3个二元组），所以精确率和召回率都是33%； 
- 用rouge-l衡量时，predictions和realitys中的最长公共子序列是`['you', 'are', 'programer']`,生成文本和参考文本的总长度都是4，所以精确率和召回率都是75%；  

下面再用中文测试。

In [12]:
predictions = "你是一个程序员。"
realitys = "你就是那个程序员。"
scores = rouge_score(predictions, realitys)
scores

{'rouge-1': {'r': 0.0, 'p': 0.0, 'f': 0.0},
 'rouge-2': {'r': 0.0, 'p': 0.0, 'f': 0.0},
 'rouge-l': {'r': 0.0, 'p': 0.0, 'f': 0.0}}

所有指标的精确率、召回率、f1-score都是0。
> 原因在于：rouge默认只支持按英文标准的空格符号来进行分词，由于上面的中文文本中没有空格，所以被当成一个词来进行重叠度计算，没有任何重叠所以输出是0。

需要引入jieba中文分词库，并封装一个tokenize方法专门用于对文本分词。

In [2]:
import jieba

def tokenize(text):
    return ' '.join(list(jieba.cut(text, cut_all=False)))

> 上面的tokenize方法中，先使用jieba.cut方法进行分词，再将分词后的序列用空格连接，转换成与英文文本相同的分隔符，以便rouge库能自动分词。

测试分词效果。

In [16]:
tokenized_predictions = tokenize(predictions)
tokenized_realitys = tokenize(realitys)
tokenized_predictions, tokenized_realitys

('你 是 一个 程序员 .', '你 就是 那个 程序员 .')

> 注：jieba分词的详细介绍和用法，请参考：[jieba分词初探](https://golfxiao.blog.csdn.net/article/details/139304506)

对分词后的中文序列计算rouge分数。

In [17]:
scores = rouge_score(tokenized_predictions, tokenized_realitys)
scores

{'rouge-1': {'r': 0.5, 'p': 0.5, 'f': 0.4999999950000001},
 'rouge-2': {'r': 0.0, 'p': 0.0, 'f': 0.0},
 'rouge-l': {'r': 0.5, 'p': 0.5, 'f': 0.4999999950000001}}

> 4个词中有`你`, `程序员`两个词相同，单词重叠率为50%，所以rouge-1的分数为50%。

 上面的例子中能够看出，rouge-1和rouge-l的两个指标更贴合我们的场景，但这两个指标有何不同呢？我们换一个例子来看。

In [21]:
tokenized_predictions = tokenize("他首先是一名老师，然后是一名程序员。")
tokenized_realitys = tokenize("他首先是一名程序员，然后是一名老师。")
print(tokenized_predictions, "\n", tokenized_realitys)
scores = rouge_score(tokenized_predictions, tokenized_realitys)
scores

他 首先 是 一名 老师 ， 然后 是 一名 程序员 。 
 他 首先 是 一名 程序员 ， 然后 是 一名 老师 。


{'rouge-1': {'r': 1.0, 'p': 1.0, 'f': 0.999999995},
 'rouge-2': {'r': 0.7777777777777778,
  'p': 0.7777777777777778,
  'f': 0.7777777727777778},
 'rouge-l': {'r': 0.7777777777777778,
  'p': 0.7777777777777778,
  'f': 0.7777777727777778}}

> 在上面这个例子中，预测序列和参考序列的单词完全相同，只是顺序不同，rouge-1不考虑单词顺序，最长子序列`['他', '首先', '是', '一名', '老师', '然后', '是', '一名', '程序员']`囊括了所有的词, 所以得出的精确率和召回率均是100%；而rouge-L关注了单词的顺序，所以最长子序列`['他', '首先', '是', '一名', '然后', '是', '一名']`只有7个词，而参考序列和预测序列的长度是9个词，所以精确率和召回率为77.7%。

由此可见，如果不考虑序列中词的顺序，则选择rouge-1，反之，如果考虑序列中词的顺序，则选择rouge-L。实际场景中，文本的顺序一般都会影响文本所表达的含义，所以为了尽可能准确，我们的场景选择rouge-L。


## 脚本改造

首先，引入原始的评测脚本`evaluate.py`。

In [3]:
%run evaluate.py



封装用于计算中文rouge分数的方法`rouge_zh_score`, 它接受两个文本串集合pred_dataset和real_dataset作为参数，返回集合中两两序列相似度的平均分数。

In [4]:
def rouge_zh_score(pred_dataset, real_dataset):
    # 首先，去掉两个集合中的空串，
    filtered_pairs = [(p, r) for p, r in zip(pred_dataset, real_dataset) if p.strip() != "" and r.strip() != ""]  
    filtered_predictions, filtered_realitys = zip(*filtered_pairs) if filtered_pairs else ([], [])  
    # 进行中文分词，因为Rouge分数的计算依赖于分词后的单词序列。
    predictions = [tokenize(text) for text in filtered_predictions]
    realitys = [tokenize(text) for text in filtered_realitys]
    # 计算两个序列的rouge分数
    score = rouge_score(predictions, realitys)
    return score['rouge-l']

> 注1：两个集合是按下标两两配对的，当出现空串时需要在两个集合中同时去掉相应位置的元素，目的是保持两个集合的配对关系不变。`zip`方法的作用是将两个集合两两配对，空串去完后还需要用`*`操作符将配对的集合解包还原为两个集合。
> 注2：需要去掉空串的原因在于，在计算 ROUGE 分数时，当任意一方字符串为空，将会导致“Hypothesis is empty.”的错误。

测试下rouge_zh_score方法的效果。

In [25]:
predictions = ["你是一个程序员。", "他首先是一名老师，然后是一名程序员。"]
realitys = ["你就是那个程序员。", "他首先是一名程序员，然后是一名老师。"]
rouge_zh_score(predictions, realitys)

{'r': 0.6888888888888889, 'p': 0.6888888888888889, 'f': 0.6888888838888889}

> 通过上面测试可以发现，`rouge.get_scores`方法既支持字符串形式，也支持集合形式，当传的是一个list集合时，它会自动计算集合中所有文本相似度分数的平均值，这是通过参数`avg=True`来指定的。

下面改造批量评测的run_test_batch方法，返回的结果不再仅仅是labels集合，还包括了speakers和reasons集合。

In [5]:
from typing import List, Dict

def make_str(data):  
    return ','.join(map(str, data)) if isinstance(data, list) else data 

def make_results(dataset):
    return {
        "labels": [item.get('label', item.get('is_fraud', False)) for item in dataset],
        "speakers": [make_str(item.get('fraud_speaker', '')) for item in dataset],
        "reasons": [item.get('reason', '') for item in dataset],
    }


def run_test_batch_v2(model, tokenizer, test_data: List[Dict], batch_size: int = 8, device='cuda'):
    real_outputs, pred_outputs = [], []
    instruction = test_data[0].get('instruction', None)
    pbar = tqdm(total=len(test_data), desc=f'progress')
    
    for i in range(0, len(test_data), batch_size):
        batch_data = test_data[i:i + batch_size]
        batch_inputs = [item['input'] for item in batch_data]
        predictions = predict_batch(model, tokenizer, batch_inputs, device, instruction=instruction)
        
        real_outputs.extend(batch_data)
        pred_outputs.extend(predictions)
        
        pbar.update(len(batch_data))

    pbar.close()
    return make_results(real_outputs), make_results(pred_outputs)

扩展一个evaluate_v2方法，此方法中区分字段走不同的指标计算：
- is_fraud: 通过sklean包的混淆矩阵来计算精确率和召回率；
- fraud_speaker: 使用sklearn包提供的accuracy_score方法来计算准确率； 
- reason: 以上面封装的中文rouge分数作为评测指标； 

In [6]:
from sklearn.metrics import accuracy_score 

def evaluate_with_model_v2(model, tokenizer, testdata_path, device='cuda', debug=False):
    dataset = load_jsonl(testdata_path)
    real_result, pred_result = run_test_batch_v2(model, tokenizer, dataset, device=device)
    
    precision, recall, accuracy = precision_recall(real_result["labels"], pred_result["labels"], debug=debug)
    print(f"is_fraud字段指标: \nprecision: {precision}, recall: {recall}, accuracy: {accuracy}")
    
    accuracy = accuracy_score(real_result["speakers"], pred_result["speakers"])  
    print(f"fraud_speaker字段指标: \naccuracy: {accuracy}")

    score = rouge_zh_score(pred_result["reasons"], real_result["reasons"])
    print(f"reason字段指标: \nprecision: {score['p']}, recall: {score['r']}, f1-score: {score['f']}")
    
    return real_result, pred_result
    

def evaluate_v2(model_path, checkpoint_path, testdata_path, device='cuda', debug=False):    
    model, tokenizer = load_model(model_path, checkpoint_path, device)
    return evaluate_with_model_v2(model, tokenizer, testdata_path, device, debug)

## 基座模型测评

先对基座模型进行评测，后续则以基座模型的评测结果作为后续微调效果的评判依据。

In [7]:
os.environ["CUDA_VISIBLE_DEVICES"] = "0"
device = 'cuda'
testdata_path = '/data2/anti_fraud/dataset/test0902.jsonl'
model_path = '/data2/anti_fraud/models/modelscope/hub/Qwen/Qwen2-1___5B-Instruct'
real_result, pred_result = evaluate_v2(model_path, '', testdata_path, device, debug=True)

progress:   3%|▎         | 96/3031 [00:39<21:26,  2.28it/s]

invalid json: 不存在经济诈骗。
invalid json: ```json
{
  "is_fraud": false,
  "fraud_speaker": "",
  "reason": ""
}
```

以上分析结果表明，该对话中不存在明显的经济诈骗行为。发言者4的回答主要涉及了美国沙发行业的趋势和商业模式的变化，并没有涉及到任何欺诈或非法活动。因此，可以认为这是一个正常的商业讨论。


progress:   4%|▍         | 120/3031 [00:47<17:25,  2.78it/s]

invalid json: 不存在经济诈骗。


progress:   4%|▍         | 128/3031 [00:50<17:44,  2.73it/s]

invalid json: 不存在经济诈骗。


progress:   5%|▍         | 144/3031 [00:55<16:08,  2.98it/s]

invalid json: 不存在经济诈骗。
invalid json: 不存在经济诈骗。


progress:   6%|▌         | 168/3031 [01:02<15:14,  3.13it/s]

invalid json: 不存在经济诈骗。


progress:   6%|▋         | 192/3031 [01:11<15:39,  3.02it/s]

invalid json: 不存在经济诈骗

分析理由：该对话中没有提到任何与经济诈骗相关的信息或行为。


progress:   7%|▋         | 200/3031 [01:13<15:19,  3.08it/s]

invalid json: 不存在经济诈骗。
invalid json: 不存在经济诈骗。


progress:   7%|▋         | 208/3031 [01:16<14:45,  3.19it/s]

invalid json: 不存在经济诈骗。


progress:   7%|▋         | 216/3031 [01:18<15:11,  3.09it/s]

invalid json: 不存在经济诈骗。


progress:   7%|▋         | 224/3031 [01:21<15:21,  3.05it/s]

invalid json: 不存在经济诈骗。
invalid json: ```json
{
  "is_fraud": false,
  "fraud_speaker": "",
  "reason": ""
}
```

经过分析，上述对话中不存在明显的经济诈骗行为。所有发言人都在讨论公司的财务状况和投资策略，并没有涉及到欺诈或非法活动。因此，可以得出结论：不存在经济诈骗。


progress:   8%|▊         | 240/3031 [01:26<15:25,  3.01it/s]

invalid json: 不存在经济诈骗

分析理由：该对话中没有提到任何关于经济诈骗的信息或行为，因此可以判断不存在经济诈骗。
invalid json: ```json
{
  "is_fraud": false,
  "fraud_speaker": "",
  "reason": ""
}
```

这个对话中没有发现任何经济诈骗的行为。姚某和张某都在介绍他们的公司和产品，并且他们提到的返利机制是合理的，符合商业伦理。因此，我们可以得出结论：这个对话不存在经济诈骗。
invalid json: 不存在经济诈骗

分析理由：对话中没有提到任何与经济诈骗相关的信息或行为。


progress:   8%|▊         | 248/3031 [01:29<14:56,  3.10it/s]

invalid json: 不存在经济诈骗。


progress:   9%|▊         | 264/3031 [01:34<15:46,  2.92it/s]

invalid json: ```json
{
  "is_fraud": false,
  "fraud_speaker": "",
  "reason": ""
}
```

这个对话中没有发现任何可疑的行为或线索，因此不存在经济诈骗的情况。对话双方都在积极地交流自己的兴趣爱好，并且讨论了理财方面的话题，这并不涉及欺诈行为。


progress:   9%|▉         | 272/3031 [01:37<15:54,  2.89it/s]

invalid json: ```json
{
  "is_fraud": false,
  "fraud_speaker": "",
  "reason": ""
}
```

经过分析，该对话中不存在明显的经济诈骗行为。发言人的陈述主要涉及养殖业的成本控制和优化策略，并未涉及到欺诈或非法活动。因此，可以判定此对话不存在经济诈骗行为。


progress:   9%|▉         | 280/3031 [01:40<15:48,  2.90it/s]

invalid json: ```json
{
  "is_fraud": false,
  "fraud_speaker": "",
  "reason": ""
}
```

经过分析，该对话中不存在明显的经济诈骗行为。所有发言者的言论都是在讨论关于科创板和全面注册制改革的问题，并没有涉及任何欺诈或非法活动。因此，我们可以得出结论：该对话不存在经济诈骗行为。


progress:  10%|▉         | 296/3031 [01:46<15:37,  2.92it/s]

invalid json: 不存在经济诈骗(is_fraud:False)，进行诈骗行为的发言者姓名(fraud_speaker:None)。

分析理由：该发言者的言论并没有涉及到任何与经济诈骗相关的活动或交易，也没有提到任何欺诈行为。因此，可以排除经济诈骗的可能性。


progress:  10%|█         | 304/3031 [01:50<18:33,  2.45it/s]

invalid json: 不存在经济诈骗。


progress:  11%|█         | 328/3031 [02:01<21:14,  2.12it/s]

invalid json: 不存在经济诈骗。


progress:  12%|█▏        | 352/3031 [02:10<17:33,  2.54it/s]

invalid json: 不存在经济诈骗。


progress:  12%|█▏        | 368/3031 [02:16<16:32,  2.68it/s]

invalid json: ```json
{
  "is_fraud": false,
  "fraud_speaker": "",
  "reason": ""
}
```

经过分析，该对话中没有发现任何经济诈骗的行为。所有发言者的言论都是关于当前疫情下的建筑行业状况和劳动力市场的讨论，并且这些信息与实际情况相符。因此，可以得出结论：不存在经济诈骗行为。


progress:  13%|█▎        | 384/3031 [02:21<15:01,  2.94it/s]

invalid json: ```json
{
    "is_fraud": false,
    "fraud_speaker": "",
    "reason": ""
}
```

经过分析，该对话中不存在明显的经济诈骗行为。发言人的言论主要围绕发电厂的运营和盈利模式进行讨论，并未涉及欺诈或非法交易。因此，可以判定此对话不存在经济诈骗行为。


progress:  13%|█▎        | 392/3031 [02:23<14:43,  2.99it/s]

invalid json: 不存在经济诈骗。


progress:  13%|█▎        | 400/3031 [02:26<14:42,  2.98it/s]

invalid json: 不存在经济诈骗。


progress:  16%|█▌        | 480/3031 [02:53<14:06,  3.01it/s]

invalid json: 不存在经济诈骗。


progress:  17%|█▋        | 504/3031 [03:01<14:21,  2.93it/s]

invalid json: 不存在经济诈骗。
invalid json: 不存在经济诈骗。


progress:  20%|█▉        | 592/3031 [03:32<13:02,  3.12it/s]

invalid json: 不存在经济诈骗。


progress:  20%|██        | 616/3031 [03:39<12:49,  3.14it/s]

invalid json: 不存在经济诈骗。
invalid json: 不存在经济诈骗。


progress:  21%|██        | 624/3031 [03:42<13:00,  3.08it/s]

invalid json: ```json
{
  "is_fraud": false,
  "fraud_speaker": "",
  "reason": ""
}
```

经过分析，该对话内容不存在经济诈骗的行为。发言者的言论都是关于酒店行业的市场趋势和经营策略，并没有涉及到欺诈行为。


progress:  21%|██        | 632/3031 [03:45<12:57,  3.09it/s]

invalid json: 不存在经济诈骗。


progress:  21%|██▏       | 648/3031 [03:50<13:35,  2.92it/s]

invalid json: 不存在经济诈骗。


progress:  22%|██▏       | 656/3031 [03:53<13:02,  3.04it/s]

invalid json: 不存在经济诈骗。


progress:  23%|██▎       | 704/3031 [04:10<13:43,  2.83it/s]

invalid json: 不存在经济诈骗。


progress:  23%|██▎       | 712/3031 [04:13<13:48,  2.80it/s]

invalid json: ```json
{
    "is_fraud": false,
    "fraud_speaker": "",
    "reason": ""
}
```

经过分析，该对话中不存在明显的经济诈骗行为。所有发言者的言论都围绕着特斯拉Model 3与特斯拉Model Y之间在市场上的竞争展开讨论，并没有涉及到任何欺诈或非法活动。因此，可以确定此对话不存在经济诈骗行为。


progress:  24%|██▍       | 728/3031 [04:19<13:50,  2.77it/s]

invalid json: 不存在经济诈骗。


progress:  25%|██▍       | 744/3031 [04:23<12:31,  3.04it/s]

invalid json: 不存在经济诈骗。


progress:  25%|██▍       | 752/3031 [04:27<13:21,  2.84it/s]

invalid json: 不存在经济诈骗。
invalid json: ```json
{
  "is_fraud": false,
  "fraud_speaker": "",
  "reason": ""
}
```

经过分析，该对话中没有发现任何涉及经济诈骗的行为。所有发言者的言论都与饲料销售、利润以及产品结构等实际业务相关的信息相符，不存在欺诈或误导性陈述的情况。因此，可以判定此对话不存在经济诈骗行为。


progress:  25%|██▌       | 768/3031 [04:33<13:48,  2.73it/s]

invalid json: 不存在经济诈骗。


progress:  26%|██▌       | 784/3031 [04:39<14:10,  2.64it/s]

invalid json: 不存在经济诈骗。


progress:  26%|██▌       | 792/3031 [04:42<13:29,  2.77it/s]

invalid json: 不存在经济诈骗。


progress:  26%|██▋       | 800/3031 [04:45<13:25,  2.77it/s]

invalid json: 不存在经济诈骗。


progress:  29%|██▊       | 864/3031 [05:08<12:31,  2.88it/s]

invalid json: ```json
{
  "is_fraud": false,
  "fraud_speaker": "",
  "reason": ""
}
```

经过分析，该对话中没有发现任何明显的经济诈骗行为。所有发言者的言论都围绕着技术问题展开，并且没有提到任何非法活动或欺诈行为。因此，我们可以得出结论：不存在经济诈骗行为。


progress:  29%|██▉       | 872/3031 [05:10<11:51,  3.04it/s]

invalid json: 不存在经济诈骗。


progress:  29%|██▉       | 888/3031 [05:16<11:33,  3.09it/s]

invalid json: 不存在经济诈骗。


progress:  31%|███       | 928/3031 [05:29<11:30,  3.05it/s]

invalid json: 不存在经济诈骗。


progress:  32%|███▏      | 968/3031 [05:43<11:26,  3.01it/s]

invalid json: 不存在经济诈骗。


progress:  32%|███▏      | 976/3031 [05:45<11:09,  3.07it/s]

invalid json: 不存在经济诈骗。


progress:  33%|███▎      | 992/3031 [05:51<11:46,  2.89it/s]

invalid json: 不存在经济诈骗

分析理由：该对话中没有涉及到任何与金钱、财产或商业交易相关的内容，因此无法判断是否为经济诈骗。


progress:  33%|███▎      | 1000/3031 [05:54<11:29,  2.95it/s]

invalid json: 不存在经济诈骗

分析理由：该对话中没有提到任何关于金钱或财务欺诈的行为，也没有涉及到任何非法活动或不道德的行为。因此，可以判断此对话不存在经济诈骗。


progress:  35%|███▍      | 1048/3031 [06:09<10:22,  3.19it/s]

invalid json: 不存在经济诈骗。
invalid json: 不存在经济诈骗。


progress:  35%|███▌      | 1064/3031 [06:14<10:15,  3.20it/s]

invalid json: 不存在经济诈骗。
invalid json: 不存在经济诈骗。


progress:  35%|███▌      | 1072/3031 [06:17<10:20,  3.16it/s]

invalid json: ```json
{
  "is_fraud": false,
  "fraud_speaker": "",
  "reason": ""
}
```

以上分析认为该对话中不存在明显的经济诈骗行为。虽然讨论涉及到了一些关于AI创作音乐等话题，但并未直接提及任何欺诈行为或利益冲突。因此，可以判定为非诈骗情况。


progress:  36%|███▌      | 1080/3031 [06:19<09:49,  3.31it/s]

invalid json: 不存在经济诈骗。
invalid json: 不存在经济诈骗。


progress:  36%|███▌      | 1088/3031 [06:22<10:21,  3.13it/s]

invalid json: ```json
{
  "is_fraud": false,
  "fraud_speaker": "",
  "reason": ""
}
```

经过分析，该对话内容不存在经济诈骗的行为。发言人3提到的数据恢复情况良好，并且对汉庭品牌的评价也表明其产品力并不属于经济型范畴。因此，可以排除经济诈骗的可能性。


progress:  37%|███▋      | 1112/3031 [06:32<11:47,  2.71it/s]

invalid json: 不存在经济诈骗。


progress:  38%|███▊      | 1144/3031 [06:43<10:36,  2.96it/s]

invalid json: 不存在经济诈骗。


progress:  38%|███▊      | 1152/3031 [06:45<10:37,  2.95it/s]

invalid json: 不存在经济诈骗。


progress:  39%|███▊      | 1168/3031 [06:51<10:26,  2.98it/s]

invalid json: ```json
{
  "is_fraud": false,
  "fraud_speaker": "",
  "reason": ""
}
```

经过分析，该对话中不存在明显的经济诈骗行为。发言人的言论主要是关于金证科技的核心业务和经营状况，并未涉及任何欺诈或非法活动。因此，可以判定此对话不存在经济诈骗行为。


progress:  39%|███▉      | 1176/3031 [06:53<10:15,  3.01it/s]

invalid json: 不存在经济诈骗。


progress:  39%|███▉      | 1184/3031 [06:56<10:10,  3.02it/s]

invalid json: 不存在经济诈骗。
invalid json: 不存在经济诈骗。


progress:  39%|███▉      | 1192/3031 [06:58<09:49,  3.12it/s]

invalid json: ```json
{
  "is_fraud": false,
  "fraud_speaker": "",
  "reason": ""
}
```

这个对话中没有发现任何可疑的行为或信息，因此可以判断这不是一个经济诈骗案例。


progress:  40%|███▉      | 1200/3031 [07:01<09:59,  3.06it/s]

invalid json: 不存在经济诈骗。
invalid json: 不存在经济诈骗。


progress:  40%|███▉      | 1208/3031 [07:03<09:02,  3.36it/s]

invalid json: 不存在经济诈骗。
invalid json: 不存在经济诈骗。
invalid json: 不存在经济诈骗。


progress:  40%|████      | 1216/3031 [07:06<09:18,  3.25it/s]

invalid json: 不存在经济诈骗。


progress:  40%|████      | 1224/3031 [07:09<10:21,  2.91it/s]

invalid json: {
  "is_fraud": true,
  "fraud_speaker": "李小龙",
  "reason": "李小龙在与张女士的对话中多次请求张女士加入他的QQ群和理赔群，这明显是为了获取她的个人信息和进一步实施诈骗。",
}


progress:  41%|████▏     | 1256/3031 [07:20<10:12,  2.90it/s]

invalid json: 不存在经济诈骗。


progress:  42%|████▏     | 1272/3031 [07:27<11:47,  2.49it/s]

invalid json: ```json
{
  "is_fraud": false,
  "fraud_speaker": "",
  "reason": ""
}
```

经过分析，该对话中不存在明显的经济诈骗行为。发言人的言论主要是关于医疗市场的政策和趋势，以及对不同产品（如OK镜）的预测。这些讨论与经济诈骗通常涉及的金融欺诈、非法传销等主题不符。因此，可以认为该对话不存在经济诈骗行为。


progress:  42%|████▏     | 1280/3031 [07:30<11:15,  2.59it/s]

invalid json: {"is_fraud":true,"reason":"该对话中可能存在经济诈骗行为，因为发言者描述了不同季节的服装尺寸和重量，暗示了某种商品或服务的价格和质量，这可能是为了诱骗消费者购买。”}


progress:  43%|████▎     | 1312/3031 [07:44<12:21,  2.32it/s]

invalid json: 不存在经济诈骗。


progress:  44%|████▍     | 1328/3031 [07:50<11:49,  2.40it/s]

invalid json: 不存在经济诈骗。


progress:  45%|████▍     | 1352/3031 [08:01<12:08,  2.31it/s]

invalid json: 不存在经济诈骗

分析理由：该对话中没有提到任何与经济诈骗相关的信息或行为。


progress:  45%|████▌     | 1368/3031 [08:10<13:39,  2.03it/s]

invalid json: ```json
{
  "is_fraud": false,
  "fraud_speaker": "",
  "reason": ""
}
```

经过分析，该对话中不存在明显的经济诈骗行为。发言人的言论主要围绕硅材料的性能、使用场景和优化策略展开，并未涉及欺诈或误导性信息。因此，可以判定此对话不存在经济诈骗行为。
invalid json: 不存在经济诈骗。


progress:  46%|████▌     | 1384/3031 [08:15<10:57,  2.50it/s]

invalid json: 不存在经济诈骗。


progress:  46%|████▌     | 1392/3031 [08:18<10:29,  2.60it/s]

invalid json: ```json
{
    "is_fraud": false,
    "fraud_speaker": "",
    "reason": ""
}
```

经过分析，该对话中不存在明显的经济诈骗行为。发言者的言论主要围绕旅游业的变化和市场需求的情况进行讨论，没有涉及到欺诈或非法活动。因此，可以判断此对话不存在经济诈骗行为。


progress:  46%|████▌     | 1400/3031 [08:21<10:15,  2.65it/s]

invalid json: 不存在经济诈骗。


progress:  47%|████▋     | 1432/3031 [08:33<10:45,  2.48it/s]

invalid json: 不存在经济诈骗。
invalid json: ```json
{
  "is_fraud": true,
  "fraud_speaker": "张俊",
  "reason": "张俊在试图向翁丽推销一个名为'财富之路'的投资平台，并声称其年化收益率高达20%，这明显违反了金融行业的正常投资回报率标准。此外，张俊还暗示投资者可以通过简单操作获得高收益，这也与正常的市场运作方式不符。因此，可以判断张俊的行为可能存在经济诈骗的可能性。"
}
```
请注意，上述分析基于虚构的情景和逻辑推断，实际情况中诈骗行为需要通过法律手段来确定。


progress:  48%|████▊     | 1440/3031 [08:36<10:38,  2.49it/s]

invalid json: ```json
{
  "is_fraud": false,
  "fraud_speaker": "",
  "reason": ""
}
```

经过分析，该对话中不存在明显的经济诈骗行为。发言人的言论主要围绕当前经济形势和电力消费数据进行讨论，并没有涉及到欺诈或非法活动。因此，可以判断此对话不存在经济诈骗行为。


progress:  49%|████▉     | 1480/3031 [08:49<08:24,  3.07it/s]

invalid json: 不存在经济诈骗。
invalid json: 不存在经济诈骗。


progress:  49%|████▉     | 1496/3031 [08:54<08:37,  2.96it/s]

invalid json: ```json
{
  "is_fraud": false,
  "fraud_speaker": "",
  "reason": ""
}
```

经过分析，该对话中不存在明显的经济诈骗行为。发言人的讨论主要围绕比亚迪2023年的汽车销售目标进行，没有涉及到欺诈或非法活动。因此，可以确定此对话不存在经济诈骗行为，并且发言者的讨论也符合正常商业交流和预测的逻辑。


progress:  50%|█████     | 1520/3031 [09:02<08:33,  2.94it/s]

invalid json: 很抱歉，我无法理解您的问题。请您提供更多的信息或者重新描述您的问题，以便于我更好地帮助您。


progress:  51%|█████▏    | 1560/3031 [09:20<09:41,  2.53it/s]

invalid json: 不存在经济诈骗。


progress:  52%|█████▏    | 1576/3031 [09:27<11:02,  2.20it/s]

invalid json: ```json
{
  "is_fraud": false,
  "fraud_speaker": "",
  "reason": ""
}
```

经过分析，该对话中不存在明显的经济诈骗行为。参与者都在探讨AI生成内容（AIGC）的相关法律和技术问题，并没有涉及任何非法或欺诈性的活动。因此，我们可以得出结论：该对话不包含经济诈骗行为。
invalid json: ```json
{
  "is_fraud": false,
  "fraud_speaker": "",
  "reason": ""
}
```

以上分析结果表明，对话中没有发现明显的经济诈骗行为。发言者5在讨论项目估值和收益时使用了一种常见的财务计算方法，即从整体估值中减去净资产成本来估算单个项目或整个项目的收益。这种做法通常用于评估投资项目的风险和潜在回报，并不涉及欺诈行为。因此，可以认为该对话不存在经济诈骗的情况。


progress:  53%|█████▎    | 1608/3031 [09:38<08:42,  2.72it/s]

invalid json: 不存在经济诈骗。


progress:  53%|█████▎    | 1616/3031 [09:41<08:11,  2.88it/s]

invalid json: 不存在经济诈骗。


progress:  54%|█████▍    | 1648/3031 [09:53<08:12,  2.81it/s]

invalid json: 不存在经济诈骗。


progress:  55%|█████▍    | 1656/3031 [09:56<08:18,  2.76it/s]

invalid json: 不存在经济诈骗。


progress:  56%|█████▌    | 1704/3031 [10:11<07:27,  2.97it/s]

invalid json: 不存在经济诈骗。
invalid json: 不存在经济诈骗。
invalid json: ```json
{
    "is_fraud": false,
    "fraud_speaker": "",
    "reason": ""
}
```

这个对话中没有发现任何经济诈骗的行为。所有的发言人都提到了他们对当前经济形势的看法和预测，并且提供了具体的例子来支持他们的观点。因此，我们可以得出结论：不存在经济诈骗的情况。


progress:  56%|█████▋    | 1712/3031 [10:14<07:01,  3.13it/s]

invalid json: 不存在经济诈骗。
invalid json: 不存在经济诈骗。


progress:  57%|█████▋    | 1728/3031 [10:19<07:14,  3.00it/s]

invalid json: ```json
{
  "is_fraud": false,
  "fraud_speaker": "",
  "reason": ""
}
```

分析结果：不存在经济诈骗。在提供的对话中，发言者4只是分享了一个关于商业需求预测的观点，并没有提到任何欺诈行为或企图欺骗他人的情况。因此，可以确定该对话不存在经济诈骗。


progress:  58%|█████▊    | 1760/3031 [10:31<07:22,  2.88it/s]

invalid json: ```json
{
  "is_fraud": false,
  "fraud_speaker": "",
  "reason": ""
}
```

分析：根据对话内容，李勇向张玲询问是否使用过“京东白条”服务，并建议她尽快注销该账户，以免影响个人征信。因此，不存在经济诈骗的行为。
invalid json: 不存在经济诈骗。
invalid json: ```json
{
  "is_fraud": false,
  "reason": ""
}
```

在上述对话中，并没有发现任何可疑的行为或迹象表明可能存在经济诈骗。所有发言者的言论都是关于电力价格的比较和解释，没有涉及到欺诈或其他非法活动。因此，可以确定不存在经济诈骗的情况。


progress:  58%|█████▊    | 1768/3031 [10:34<07:35,  2.78it/s]

invalid json: 不存在经济诈骗。


progress:  59%|█████▊    | 1776/3031 [10:37<07:44,  2.70it/s]

invalid json: 不存在经济诈骗。


progress:  59%|█████▉    | 1800/3031 [10:45<07:04,  2.90it/s]

invalid json: 不存在经济诈骗。


progress:  61%|██████    | 1856/3031 [11:04<06:58,  2.81it/s]

invalid json: 不存在经济诈骗。


progress:  62%|██████▏   | 1880/3031 [11:12<06:27,  2.97it/s]

invalid json: 不存在经济诈骗。


progress:  63%|██████▎   | 1896/3031 [11:17<06:00,  3.14it/s]

invalid json: 不存在经济诈骗。


progress:  63%|██████▎   | 1920/3031 [11:26<06:20,  2.92it/s]

invalid json: 不存在经济诈骗。


progress:  64%|██████▎   | 1928/3031 [11:28<06:02,  3.04it/s]

invalid json: 不存在经济诈骗。


progress:  64%|██████▍   | 1936/3031 [11:31<05:57,  3.06it/s]

invalid json: 不存在经济诈骗。
invalid json: ```json
{
  "is_fraud": false,
  "fraud_speaker": "",
  "reason": ""
}
```

经过分析，该对话中不存在明显的经济诈骗行为。发言人的言论主要围绕着旅游统计数据的准确性进行讨论，并提出了可能存在的误差来源。因此，可以判断此对话不存在经济诈骗行为。
invalid json: 不存在经济诈骗。


progress:  64%|██████▍   | 1952/3031 [11:37<06:24,  2.80it/s]

invalid json: 不存在经济诈骗。


progress:  65%|██████▍   | 1968/3031 [11:42<05:47,  3.06it/s]

invalid json: 不存在经济诈骗。
invalid json: 不存在经济诈骗。


progress:  65%|██████▌   | 1984/3031 [11:47<05:59,  2.92it/s]

invalid json: ```json
{
    "is_fraud": false,
    "fraud_speaker": "",
    "reason": ""
}
```

经过分析，该对话中没有发现明显的经济诈骗行为。所有发言者的言论都围绕着酒店预订的价格、地区差异以及会员制度等话题展开，并未涉及任何欺诈或非法活动。因此，我们可以得出结论：不存在经济诈骗行为。


progress:  66%|██████▌   | 2000/3031 [11:53<05:56,  2.89it/s]

invalid json: 不存在经济诈骗。


progress:  66%|██████▌   | 2008/3031 [11:56<05:54,  2.89it/s]

invalid json: 不存在经济诈骗。


progress:  67%|██████▋   | 2032/3031 [12:04<05:36,  2.97it/s]

invalid json: ```json
{
    "is_fraud": false,
    "fraud_speaker": "",
    "reason": ""
}
```

经过分析，该对话中不存在明显的经济诈骗行为。发言者4的回答主要涉及海外客户的订单情况和库存变化，并没有涉及到任何欺诈或非法活动。因此，可以判断此对话不存在经济诈骗行为。
invalid json: 不存在经济诈骗。


progress:  68%|██████▊   | 2048/3031 [12:08<05:13,  3.13it/s]

invalid json: 不存在经济诈骗。


progress:  68%|██████▊   | 2056/3031 [12:11<05:22,  3.02it/s]

invalid json: 不存在经济诈骗。
invalid json: 不存在经济诈骗

分析理由：该对话中没有提到任何与经济诈骗相关的信息或行为。


progress:  68%|██████▊   | 2072/3031 [12:16<05:15,  3.04it/s]

invalid json: 不存在经济诈骗。


progress:  70%|██████▉   | 2120/3031 [12:33<05:10,  2.94it/s]

invalid json: 不存在经济诈骗

```json
{
  "is_fraud": false,
  "fraud_speaker": "",
  "reason": ""
}
```


progress:  70%|███████   | 2136/3031 [12:39<05:17,  2.82it/s]

invalid json: 不存在经济诈骗

分析理由：该对话中没有提到任何与经济趋势、复苏主线、龙头公司等相关的内容，也没有涉及到任何欺诈或非法活动。因此，可以判断此对话不存在经济诈骗。


progress:  71%|███████   | 2144/3031 [12:42<05:38,  2.62it/s]

invalid json: 不存在经济诈骗。


progress:  72%|███████▏  | 2168/3031 [12:49<04:28,  3.21it/s]

invalid json: 不存在经济诈骗。
invalid json: 不存在经济诈骗。


progress:  73%|███████▎  | 2208/3031 [13:02<04:27,  3.08it/s]

invalid json: ```json
{
  "is_fraud": false,
  "fraud_speaker": "",
  "reason": ""
}
```

经过分析，该对话中不存在明显的经济诈骗行为。发言人的言论主要涉及财务数据和业务运营情况，并没有涉及到欺诈或非法活动。因此，可以判断此对话不存在经济诈骗行为。


progress:  74%|███████▎  | 2232/3031 [13:12<05:17,  2.52it/s]

invalid json: 不存在经济诈骗。
invalid json: ```json
{
  "is_fraud": false,
  "fraud_speaker": "",
  "reason": ""
}
```

以上分析认为该对话不存在经济诈骗行为。在对话中，发言人1详细解释了管道内部合金与双六品质的关系，并提出了通过煅烧来解决这个问题的方法。而发言人4则是在讨论当前市场环境下的双六使用情况，没有涉及到经济诈骗的问题。因此，可以判断此对话不存在经济诈骗行为。
invalid json: 不存在经济诈骗。


progress:  75%|███████▌  | 2288/3031 [13:30<04:23,  2.82it/s]

invalid json: 不存在经济诈骗。
invalid json: ```json
{
    "is_fraud": false,
    "fraud_speaker": "",
    "reason": ""
}
```

分析结果：该对话中不存在经济诈骗的行为。发言者2提供了关于制氧机海外销售比例、恢复情况以及当前市场环境的信息，而发言者1则询问了相关数据，并就未来的发展趋势进行了讨论。因此，没有发现任何欺诈行为或可疑活动。


progress:  76%|███████▌  | 2296/3031 [13:32<03:56,  3.11it/s]

invalid json: 不存在经济诈骗。


progress:  76%|███████▋  | 2312/3031 [13:38<03:57,  3.02it/s]

invalid json: 不存在经济诈骗。


progress:  77%|███████▋  | 2320/3031 [13:41<04:03,  2.92it/s]

invalid json: 不存在经济诈骗。


progress:  77%|███████▋  | 2336/3031 [13:46<03:50,  3.02it/s]

invalid json: 不存在经济诈骗

分析理由：该对话中没有提到任何与金钱或财务相关的信息，也没有涉及到任何欺诈行为。因此，可以判断此对话不存在经济诈骗。


progress:  78%|███████▊  | 2360/3031 [13:54<03:40,  3.04it/s]

invalid json: 不存在经济诈骗。


progress:  78%|███████▊  | 2368/3031 [13:57<03:37,  3.04it/s]

invalid json: 不存在经济诈骗

分析理由：该对话中没有涉及到任何与金钱、财务相关的行为或交易，也没有提到任何欺诈行为。因此，无法判断是否为经济诈骗。
invalid json: ```json
{
  "is_fraud": false,
  "fraud_speaker": "",
  "reason": ""
}
```

经过分析，该对话中不存在明显的经济诈骗行为。发言人的讨论主要围绕当前血糖监测设备市场的现状和未来发展趋势展开，并没有涉及到具体的金额或交易细节，因此无法判断是否涉及经济诈骗。


progress:  79%|███████▉  | 2392/3031 [14:04<03:26,  3.10it/s]

invalid json: 不存在经济诈骗

分析理由：这段对话中没有提到任何与经济诈骗相关的信息或行为，因此可以判断这段对话不存在经济诈骗。


progress:  79%|███████▉  | 2400/3031 [14:07<03:35,  2.93it/s]

invalid json: ```json
{
  "is_fraud": false,
  "fraud_speaker": "",
  "reason": ""
}
```

在这个对话中，没有发现任何可疑的行为或线索表明存在经济诈骗。发言人1和发言人2之间的对话似乎是在讨论关于呼吸机和制氧机的需求变化的问题，并且他们提供了基于公司的观察结果来支持自己的观点。因此，我们可以得出结论：不存在经济诈骗行为。


progress:  80%|███████▉  | 2416/3031 [14:13<03:34,  2.87it/s]

invalid json: 不存在经济诈骗。


progress:  80%|████████  | 2432/3031 [14:19<03:42,  2.69it/s]

invalid json: ```json
{
    "is_fraud": false,
    "fraud_speaker": "",
    "reason": ""
}
```

以上分析认为不存在经济诈骗行为。提供的信息主要是关于酒店行业在特定时间段内的数据变化情况，并没有涉及到任何欺诈或非法活动。


progress:  81%|████████  | 2448/3031 [14:24<03:15,  2.98it/s]

invalid json: 不存在经济诈骗。


progress:  81%|████████▏ | 2464/3031 [14:29<02:56,  3.22it/s]

invalid json: 不存在经济诈骗。


progress:  82%|████████▏ | 2472/3031 [14:31<02:53,  3.23it/s]

invalid json: 不存在经济诈骗(is_fraud:False)，正在进行诈骗行为的发言者姓名(fraud_speaker:None)。


progress:  82%|████████▏ | 2496/3031 [14:40<03:10,  2.81it/s]

invalid json: 不存在经济诈骗。


progress:  83%|████████▎ | 2528/3031 [14:51<03:03,  2.73it/s]

invalid json: 不存在经济诈骗。


progress:  84%|████████▎ | 2536/3031 [14:54<02:58,  2.77it/s]

invalid json: 不存在经济诈骗

分析理由：该对话中没有提到任何关于经济诈骗的信息或行为。


progress:  84%|████████▍ | 2544/3031 [14:57<02:52,  2.82it/s]

invalid json: 不存在经济诈骗。


progress:  85%|████████▍ | 2576/3031 [15:08<02:34,  2.94it/s]

invalid json: 不存在经济诈骗。


progress:  86%|████████▌ | 2600/3031 [15:16<02:24,  2.98it/s]

invalid json: ```json
{
    "is_fraud": false,
    "fraud_speaker": "",
    "reason": ""
}
```

经过分析，该对话内容不存在经济诈骗的行为。发言者的言论都是关于酒店行业的市场趋势和经营策略，并没有涉及到欺诈行为。


progress:  87%|████████▋ | 2624/3031 [15:26<02:45,  2.46it/s]

invalid json: 不存在经济诈骗。


progress:  87%|████████▋ | 2648/3031 [15:35<02:24,  2.66it/s]

invalid json: ```json
{
  "is_fraud": false,
  "fraud_speaker": "",
  "reason": ""
}
```

经过分析，该对话中不存在明显的经济诈骗行为。发言人的言论反映了市场情况和消费者的偏好，并未涉及欺诈或非法活动。


progress:  88%|████████▊ | 2656/3031 [15:38<02:14,  2.78it/s]

invalid json: 不存在经济诈骗。
invalid json: 不存在经济诈骗。


progress:  88%|████████▊ | 2672/3031 [15:43<02:03,  2.91it/s]

invalid json: 不存在经济诈骗。


progress:  88%|████████▊ | 2680/3031 [15:46<02:00,  2.90it/s]

invalid json: ```json
{
  "is_fraud": false,
  "fraud_speaker": "",
  "reason": ""
}
```

在这个对话中，并没有发现任何明显的经济诈骗行为。所有的发言人都在讨论与航运相关的投资策略，没有提到欺诈或非法活动。因此，我们可以得出结论：不存在经济诈骗行为。


progress:  89%|████████▊ | 2688/3031 [15:49<02:06,  2.72it/s]

invalid json: 不存在经济诈骗。


progress:  89%|████████▉ | 2696/3031 [15:52<02:06,  2.65it/s]

invalid json: ```json
{
  "is_fraud": false,
  "fraud_speaker": "",
  "reason": ""
}
```

经过分析，该对话中不存在明显的经济诈骗行为。发言人的言论都是关于税收政策和煤炭价格的解释，并没有涉及到任何非法或欺诈性的活动。因此，可以确定此对话不存在经济诈骗行为。


progress:  89%|████████▉ | 2704/3031 [15:55<01:58,  2.75it/s]

invalid json: 不存在经济诈骗。


progress:  90%|█████████ | 2736/3031 [16:06<01:39,  2.95it/s]

invalid json: ```json
{
  "is_fraud": false,
  "fraud_speaker": "",
  "reason": ""
}
```

经过分析，该对话中不存在明显的经济诈骗行为。所有发言者的言论都围绕着当前经济形势和企业生存之道进行讨论，并未涉及任何欺诈或非法活动。因此，可以判定此对话不存在经济诈骗行为。
invalid json: 不存在经济诈骗。


progress:  91%|█████████ | 2744/3031 [16:08<01:37,  2.96it/s]

invalid json: ```json
{
  "is_fraud": false,
  "fraud_speaker": "",
  "reason": ""
}
```

经过分析，该对话中不存在明显的经济诈骗行为。发言者3的回答表明了对房地产市场的预测，并没有涉及到任何欺诈行为。因此，可以判断此对话不存在经济诈骗。


progress:  91%|█████████ | 2752/3031 [16:11<01:33,  2.99it/s]

invalid json: 不存在经济诈骗。
invalid json: 不存在经济诈骗。


progress:  91%|█████████ | 2760/3031 [16:13<01:29,  3.04it/s]

invalid json: 不存在经济诈骗。


progress:  91%|█████████▏| 2768/3031 [16:17<01:36,  2.71it/s]

invalid json: 不存在经济诈骗。
invalid json: ```json
{
  "is_fraud": false,
  "fraud_speaker": "",
  "reason": ""
}
```

以上分析认为不存在经济诈骗行为。在提供的对话中，没有发现任何明显的欺诈或误导性言论。参与者讨论了关于景区免票政策对旅游业的影响，并且提到了一些与旅游目的地相关的原因，如地理位置、交通便利性和节假日效应等。这些讨论并没有涉及到欺诈性的信息或企图欺骗他人。因此，可以得出结论：不存在经济诈骗行为。


progress:  92%|█████████▏| 2792/3031 [16:26<01:24,  2.84it/s]

invalid json: 不存在经济诈骗

分析理由：该对话中没有提到任何与经济诈骗相关的信息或行为。


progress:  92%|█████████▏| 2800/3031 [16:29<01:23,  2.76it/s]

invalid json: 不存在经济诈骗。


progress:  93%|█████████▎| 2824/3031 [16:36<01:09,  2.98it/s]

invalid json: 不存在经济诈骗。


progress:  93%|█████████▎| 2832/3031 [16:41<01:19,  2.51it/s]

invalid json: 不存在经济诈骗(is_fraud:False)，进行诈骗行为的发言者姓名(fraud_speaker:None)。

分析理由：该对话内容没有涉及到任何与金钱、财产或商业交易相关的信息，因此不能判断是否为经济诈骗。
invalid json: 不存在经济诈骗。


progress:  94%|█████████▎| 2840/3031 [16:44<01:13,  2.58it/s]

invalid json: 不存在经济诈骗。
invalid json: 不存在经济诈骗

分析理由：该对话中没有提到任何与金钱或财务相关的行为，也没有涉及到任何欺诈、欺骗或非法活动。因此，可以确定该对话不存在经济诈骗。


progress:  94%|█████████▍| 2848/3031 [16:47<01:10,  2.60it/s]

invalid json: 不存在经济诈骗。


progress:  94%|█████████▍| 2856/3031 [16:50<01:11,  2.46it/s]

invalid json: 不存在经济诈骗。


progress:  96%|█████████▌| 2904/3031 [17:08<00:48,  2.64it/s]

invalid json: 不存在经济诈骗。


progress:  96%|█████████▌| 2912/3031 [17:11<00:42,  2.79it/s]

invalid json: 不存在经济诈骗。


progress:  96%|█████████▋| 2920/3031 [17:13<00:37,  2.99it/s]

invalid json: 不存在经济诈骗。


progress:  97%|█████████▋| 2936/3031 [17:20<00:35,  2.67it/s]

invalid json: 不存在经济诈骗。
invalid json: 不存在经济诈骗

分析理由：该对话中没有提到任何与金钱或财务相关的内容，也没有涉及到任何欺诈行为。因此，可以判断此对话不存在经济诈骗。


progress:  97%|█████████▋| 2944/3031 [17:22<00:30,  2.83it/s]

invalid json: 不存在经济诈骗。
invalid json: 不存在经济诈骗。


progress:  97%|█████████▋| 2952/3031 [17:28<00:35,  2.22it/s]

invalid json: ```json
{
    "is_fraud": false,
    "fraud_speaker": "",
    "reason": ""
}
```

经过分析，该对话中不存在经济诈骗的行为。发言者的言论都是基于对当前市场情况和公司策略的描述，并没有涉及到欺诈或非法活动。


progress:  98%|█████████▊| 2960/3031 [17:30<00:28,  2.49it/s]

invalid json: 由于您没有提供具体的对话内容，我无法进行分析。请提供具体的内容以便我为您分析。


progress:  99%|█████████▊| 2992/3031 [17:40<00:12,  3.01it/s]

invalid json: 不存在经济诈骗。
invalid json: ```json
{
  "is_fraud": false,
  "fraud_speaker": "",
  "reason": ""
}
```

经过分析，该对话中不存在明显的经济诈骗行为。发言人的讨论主要是关于电子血压计行业的现状和未来趋势，没有涉及到任何欺诈或非法活动。因此，可以确定此对话不存在经济诈骗行为。
invalid json: 不存在经济诈骗。


progress:  99%|█████████▉| 3008/3031 [17:45<00:07,  3.01it/s]

invalid json: ```json
{
  "is_fraud": false,
  "fraud_speaker": "",
  "reason": ""
}
```

以上分析结果表明，该对话中不存在明显的经济诈骗行为。所有发言者的言论都是关于产品的生产和销售情况的讨论，并且没有涉及到欺诈或非法活动。因此，可以得出结论：此对话不存在经济诈骗行为。


progress: 100%|██████████| 3031/3031 [17:55<00:00,  2.82it/s]
Building prefix dict from the default dictionary ...
Loading model from cache /tmp/jieba.cache


tn：590, fp:903, fn:44, tp:1494
is_fraud字段指标: 
precision: 0.623279098873592, recall: 0.9713914174252276, accuracy: 0.6875618607720224
fraud_speaker字段指标: 
accuracy: 0.6374133949191686


Loading model cost 0.776 seconds.
Prefix dict has been built successfully.


reason字段指标: 
precision: 0.33552894583785237, recall: 0.23240362412953733, f1-score: 0.2638127489291967


| 字段 | 指标  |
| --- | --- |
| is_fraud |  precision: 0.6232, recall: 0.9713, accuracy=0.6875 |
|fraud_speaker| accuracy: 0.6374 |
|reason| precision: 0.3355, recall: 0.2324, f1-score: 0.2638 |

从上面的指标来看，只有is_fraud字段的 召回率`0.9673`比较高，但其它指标（如is_fraud字段的精确率，fraud_speaker字段的准确率，reason字段的各项指标）都比较低。

**小结**：本文通过对rouge指标的调研测试，了解了rouge指标中的rouge-1和rouge-L子指标的区别，并对评测脚本进行了改造升级，以支持分类原因和欺诈者字段的生成质量评测，为后续分类原因和欺诈者字段的训练微调提供了一个效果评测的基准。

## 参考阅读
- [数据校正与增强](https://golfxiao.blog.csdn.net/article/details/142333893)
- [模型评测](https://golfxiao.blog.csdn.net/article/details/141355995)
- [批量评测改造](https://golfxiao.blog.csdn.net/article/details/141609627)
- [大模型评估指标之ROUGE](https://blog.csdn.net/BIT_666/article/details/132347794)